hudi-common/src/main/java/org/apache/hudi/common/util/collection/ExternalSpillableMap.java

```diff
@@ -92,7 +92,7 @@ public class ExternalSpillableMap<T extends Serializable, R> implements Map<T, R
   public ExternalSpillableMap(long maxInMemorySizeInBytes, String baseFilePath, SizeEstimator<T> keySizeEstimator,
                               SizeEstimator<R> valueSizeEstimator, DiskMapType diskMapType, CustomSerializer<R> valueSerializer,
                               boolean isCompressionEnabled, String loggingContext) throws IOException {
-    this.inMemoryMap = new HashMap<>();
+    this.inMemoryMap = new ConcurrentHashMap<>();
```

Comment:
Will updating this class to thread-safe cost us performance in scenarios that we used only within a single thread? For example, the `map` used to store log records in the file group reader

Reply:
I am thinking it may, also see the above thread about whether we should be fixing the issue at the FSView level. That seems like the better way and makes sure any future storage options for the FSView get the benefit. I will look into this later this week.